File: local_market_ui/lib/auth/models/jwt_token.dart
import 'package:json_annotation/json_annotation.dart';
part 'jwt_token.g.dart';

@JsonSerializable()
class JWTToken {
  
   int? id;

   String?accessToken;

   String?tokenCreationTime;

   String? tokenExpireTime;

   String? tokenType="Bearer";

   String? email;


   JWTToken(this.id, this.accessToken, this.tokenCreationTime, this.tokenExpireTime, this.tokenType, this.email);

   JWTToken.fromJson(Map<String, dynamic> json) {
      id = json['id'];
      accessToken = json['accessToken'];
      tokenCreationTime= json['tokenCreationTime'];
      tokenExpireTime= json['tokenExpireTime'];
     tokenType= json['tokenType'];
     email= json['email'];
   }

   Map<String, dynamic> toJson() {
      final Map<String, dynamic> data = <String, dynamic>{};

      data['id'] = id;
      data['accessToken'] = accessToken;
      data['tokenCreationTime'] = tokenCreationTime;
      data['tokenExpireTime'] = tokenExpireTime;
      data['tokenType'] = tokenType;
      data['email'] = email;
      return data;
   }
}

File: Apna_Mall/src/main/java/org/spring/security/entity/auth/JWTToken.java
package org.spring.security.entity.auth;


import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@Entity(name = "jwt_token")
public class JWTToken {

    @Id
    private String email;

    @Column(columnDefinition = "TEXT")
    private String accessToken;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private Date tokenCreationTime;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private Date tokenExpireTime;

    private String tokenType = "Bearer";


}
